### PR TITLE
docs: add section to 1.12 upgrade notes about new TLS config style

### PR DESCRIPTION
**Pull request by boxofrad**
### Description
Whoops, didn't get a chance to add this before the release!

### CODE REVIEW

File: website/content/docs/upgrading/upgrade-specific.mdx [hunk 1]
```diff
@@ -26,6 +26,24 @@ to `true`. In prior Consul versions (1.10.x through 1.11.x), the config defaulte
 well as the flag will be removed in upcoming Consul 1.13. We recommend changing your instrumentation to use 1.10 and later
 style `consul.api.http...` metrics and removing the configuration flag from your setup.
 
+### TLS Configuration
+
+TLS can now be configured differently on each of Consul's exposed listeners/ports.
+As such, the following top-level configuration fields are deprecated and should be
+replaced with the new `tls` stanza:
```
Comment (boxofrad) [inline website/content/docs/upgrading/upgrade-specific.mdx:33]: I'd like to link to the new TLS docs from here, but it looks like they were accidentally deleted in https://github.com/hashicorp/consul/commit/f4c03d2340d5650ba409aa3f4bfa813560d165b2 so will wait until they're restored 🙇🏻‍♂️

[the remark this answers is not in the document]
Comment (boxofrad) [reply website/content/docs/upgrading/upgrade-specific.mdx:33]: So much better, thank you! 🎉

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/670998.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1bac1544d094ef356787974cd0440783142e53e1 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1bac1544d094ef356787974cd0440783142e53e1 onto `release/1.12.x` succeeded!

**Pull request merged**